web-marketplace/src/components/organisms/Order/Order.Summary.tsx

```diff
@@ -29,40 +26,29 @@ interface OrderSummaryProps {
   blockchainDetails: BlockchainDetailsData;
   creditsData: CreditsData;
   paymentInfo: PaymentInfoData;
-  allowedDenoms?: AllowedDenom[];
+  displayTotalPrice: number;
+  displayDenom: string;
+  currency: {
+    askDenom: string;
+    askBaseDenom: string;
+  };
 }
 
 export const OrderSummary = ({
   creditsData,
   retirementInfo,
   blockchainDetails,
   paymentInfo,
-  allowedDenoms,
+  displayTotalPrice,
+  displayDenom,
+  currency,
 }: OrderSummaryProps) => {
   const { _ } = useLingui();
   const { purchaseDate, blockchainRecord } = blockchainDetails;
   const { askDenom, askBaseDenom, cardLast4, cardBrand } = paymentInfo;
   const isCardPayment = cardLast4 && askDenom === USD_DENOM;
   const { reason, location, retiredCredits } = retirementInfo;
-  const { credits, totalPrice } = creditsData;
-
-  const denom = allowedDenoms?.find(denom => denom.bankDenom === askDenom);
-  const displayTotalPrice = denom
-    ? +totalPrice * Math.pow(10, -denom.exponent)
-    : +totalPrice;
-  const currency = {
-    askDenom,
-    askBaseDenom,
-  };
-  const displayDenom = useMemo(
-    () =>
-      findDisplayDenom({
-        allowedDenoms,
-        bankDenom: askDenom,
-        baseDenom: askBaseDenom,
-      }),
-    [allowedDenoms, askBaseDenom, askDenom],
-  );
+  const { credits } = creditsData;
```

Comment:
maybe we could just pass in `credits={creditsData.credits}` as prop instead of the full `creditsData` since we do not use the other properties here directly